WeakAuras/Transmission.lua

```diff
@@ -663,11 +665,26 @@ function TableToString(inTable, forChat)
 end
 
 function StringToTable(inString, fromChat)
-  -- if gsub strips off a ! at the beginning then we know that this is not a legacy encoding
-  local encoded, usesDeflate = inString:gsub("^%!", "")
+  -- encoding format:
+  -- version 0: simple b64 string, compressed with LC and serialized with AS
+  -- version 1: b64 string prepended with "!", compressed with LD and serialized with AS
+  -- version 2+: b64 string prepended with !WA:N! (where N is encode version)
+  --   compressed with LD and serialized with LS
+  local encodeVersion = 0
+  local encoded = ""
+  if inString:find("^%!") then
+    local header
+    inString:gsub("^%!WA%:%d+!", function(m) header = m return "" end)
+    if not header then
+      encodeVersion = 1
+    else
+      encodeVersion = tonumber(header:match("%d+"))
+    end
+  end
```

Comment:
`encoded` is an empty string after getting the `encodeVersion`, I suggest the following:
```lua
  local _, _, encodeVersion, encoded = inString:find("^(!WA:%d+!)(.+)$")
  if encodeVersion then 
    encodeVersion = tonumber(encodeVersion:match("%d+"))
  else
    encoded, encodeVersion = inString:gsub("^%!", "")
  end
```

Wago is ready and will decode strings with the new serialization format. Some time after the next version of WA is released and distributed I can have Wago start using the new lib when generating the strings as well.